web/src/main/java/com/garret/movies/batch/ShortMovieReader.java

```diff
@@ -34,13 +34,7 @@ void init() {
     }
 
     @Override
-    public ShortMovie read() throws Exception, UnexpectedInputException, ParseException, NonTransientResourceException {
-        ShortMovie shortMovie = null;
-
-        if (!movieData.isEmpty()) {
-            shortMovie = movieData.remove(0);
-        }
-
-        return shortMovie;
+    public ShortMovie read() {
+        return index.get() < movieData.size() ? movieData.get(index.getAndIncrement()) : null;
```

Comment:
@Garret-vin OutOfbounds exception goes here ))

Reply:
@Dominys I fixed it in last commit